### PR TITLE
feature : shared-preferences

### PR DESCRIPTION
**Pull request by danielairinei**

### CODE REVIEW

File: app/src/main/java/com/internship/move/feature/authentification/register/RegisterFragment.kt [hunk 1]
```diff
@@ -15,6 +16,14 @@ class RegisterFragment : Fragment(R.layout.fragment_register) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        val sharedPref = activity?.getPreferences(Context.MODE_PRIVATE) ?: return
+
+        sharedPref.edit().putString(getString(R.string.saved_splash_next_fragment), SPLASH_TO_REGISTER).apply()
```
Comment (tapp-hadaralex) [inline app/src/main/java/com/internship/move/feature/authentification/register/RegisterFragment.kt]: Do you want to check if the user went to the splash?
If yes, you do not need to store such strings in the `strings.xml` files. This is actually a bug.

strings.xml is used for configuration changes. Those include language changes, orientation changes etc.
So let's suppose you have `SPLASH` word in the english strings.xml file, but `INTRO` in the romanian strings.xml file.  If the user changes phone's language between sessions, then this approach won't work

Comment (tapp-hadaralex) [reply app/src/main/java/com/internship/move/feature/authentification/register/RegisterFragment.kt]: You could use in this case a `boolean` to see if you need to show a screen again or not.
Or if you could even a static string, like `SPLASH_TO_REGISTER` directly

File: app/src/main/java/com/internship/move/feature/splash/SplashFragment.kt [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        val sharedPref = activity?.getPreferences(Context.MODE_PRIVATE)
+
         Handler(Looper.getMainLooper()).postDelayed(
-            { findNavController().navigate(SplashFragmentDirections.actionSplashFragmentToOnboardingGraph()) },
+            {
+                when (sharedPref?.getString(
```
Comment (tapp-hadaralex) [inline app/src/main/java/com/internship/move/feature/splash/SplashFragment.kt:24]: `when` is a bit confusing. Don't use resource strings in this case.

See my comment from above and change it.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -11,4 +11,6 @@
     <string name="onboarding_rules_tutorial">You must be 18 years or and older with a valid driving licence to operate a scooter. Please follow all street signs, signals and markings, and obey local traffic laws.</string>
     <string name="onboarding_next_button">Next</string>
     <string name="onboarding_get_started_button">Get Started</string>
+    <string name="saved_splash_next_fragment">Splash Next Fragment</string>
```
Comment (tapp-hadaralex) [inline app/src/main/res/values/strings.xml]: strings.xml is strictly only for configuration changes strings.

**Review by tapp-hadaralex — CHANGES_REQUESTED** [review]
Even though it is not related to this project, there are some bugs worth noticing:

- on smaller screen phones, the Next buttons overlaps with the text.
- once I Skip the intro, I can't go back, no matter how many times I press "Log out"

File: app/src/main/java/com/internship/move/feature/authentification/register/RegisterFragment.kt (hunk 1 above)
Comment (danielairinei) [reply app/src/main/java/com/internship/move/feature/authentification/register/RegisterFragment.kt]: saved_splash_next_fragment is just a string that represents the key for the sharedPref, it's not something that's visible in the app

File: app/src/main/java/com/internship/move/feature/splash/SplashFragment.kt (hunk 2 above)
Comment (danielairinei) [reply app/src/main/java/com/internship/move/feature/splash/SplashFragment.kt:24]: Cata told me to use when in this case 😄

**Pull request merged**